src/main/java/org/openstreetmap/atlas/checks/validation/intersections/EdgeCrossingEdgeCheck.java

```diff
@@ -64,47 +71,14 @@ private static boolean canCross(final PolyLine edgeAsPolyLine, final Optional<Lo
                 // Otherwise, if crossing edges has valid, but different tag values
                 // Then that is still a valid crossing
                 || edgeLayer.isPresent() && crossingEdgeLayer.isPresent()
-                        && !edgeLayer.orElse(0L).equals(crossingEdgeLayer.orElse(0L));
-    }
-
-    private static String generateAtlasObjectPairIdentifier(final AtlasObject thisObject,
-            final AtlasObject thatObject)
-    {
-        if (thisObject.getIdentifier() < thatObject.getIdentifier())
-        {
-            return thisObject.getIdentifier() + CommonConstants.DASH + thatObject.getIdentifier();
-        }
-
-        return thatObject.getIdentifier() + CommonConstants.DASH + thisObject.getIdentifier();
-    }
-
-    /**
-     * Validates given {@link AtlasObject} (assumed to be an {@link Edge}) whether it is a valid
-     * crossing edge or not
-     *
-     * @param object
-     *            {@link AtlasObject} to test
-     * @return {@code true} if given {@link AtlasObject} object is a valid crossing edge
-     */
-    private static boolean isValidCrossingEdge(final AtlasObject object)
-    {
-        if (Edge.isMasterEdgeIdentifier(object.getIdentifier())
-                && !TagPredicates.IS_AREA.test(object))
-        {
-            final Optional<HighwayTag> highway = HighwayTag.highwayTag(object);
-            if (highway.isPresent())
-            {
-                return HighwayTag.isCarNavigableHighway(highway.get())
-                        && !HighwayTag.CROSSING.equals(highway.get());
-            }
-        }
-
-        return false;
+                        && !edgeLayer.get().equals(crossingEdgeLayer.get());
     }
 
     public EdgeCrossingEdgeCheck(final Configuration configuration)
     {
         super(configuration);
+        this.minimumHighwayType = configurationValue(configuration, "minimum.highway.type",
```

Comment:
There can sometimes be a lot of lower classification items that sort of obfuscate the results. The idea was to add the configurable so that you could focus on more important roads first. However, the lower classifications are still not supposed to cross. Perhaps it would make sense to have the configurable but set its default to the lowest highway classification?

Reply:
Yeah that would be good.